paymentsheet/src/main/java/com/stripe/android/link/ui/LinkLoadingScreen.kt

```diff
@@ -0,0 +1,67 @@
+package com.stripe.android.link.ui
+
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.heightIn
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.size
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.CompositionLocalProvider
+import androidx.compose.runtime.compositionLocalOf
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.LocalDensity
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.DpSize
+import androidx.compose.ui.unit.IntSize
+import androidx.compose.ui.unit.dp
+import com.stripe.android.link.theme.DefaultLinkTheme
+
+private val LocalLinkScreenSize = compositionLocalOf<DpSize?> { null }
+
+/**
+ * Displays a loading spinner in the center of the screen.
+ * Will match the last screen size using [ProvideLinkScreenSize] to minimize size changes.
+ */
+@Composable
+internal fun LinkLoadingScreen(modifier: Modifier = Modifier) {
```

Comment:
👀 Demo of the problem below. Note how it goes from full screen → short → medium height. It now goes from full → medium height.

https://github.com/user-attachments/assets/483ea63f-4369-40e9-a7c1-52ec26ddf3b8